app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx

```diff
@@ -0,0 +1,371 @@
+/** @file Form to create a project. */
+import * as React from 'react'
+import toast from 'react-hot-toast'
+
+import * as backendModule from '../backend'
+import * as backendProvider from '../../providers/backend'
+import * as columnModule from '../column'
+import * as error from '../../error'
+import * as fileInfo from '../../fileInfo'
+import * as loggerProvider from '../../providers/logger'
+import * as modalProvider from '../../providers/modal'
+import * as svg from '../../components/svg'
+import * as toastPromiseMultiple from '../../toastPromiseMultiple'
+import * as uniqueString from '../../uniqueString'
+
+import CreateForm, * as createForm from './createForm'
+import Table, * as table from './table'
+import ConfirmDeleteModal from './confirmDeleteModal'
+import ContextMenu from './contextMenu'
+import ContextMenuEntry from './contextMenuEntry'
+import EditableSpan from './editableSpan'
+
+// =================
+// === Constants ===
+// =================
+
+/** Placeholder row when the search query is not empty. */
+const PLACEHOLDER_WITH_QUERY = (
+    <span className="opacity-75">This folder does not contain any files matching your query.</span>
+)
+/** Placeholder row when the search query is empty. */
+const PLACEHOLDER_WITHOUT_QUERY = (
+    <span className="opacity-75">This folder does not contain any files.</span>
+)
+/** Messages to be passed to {@link toastPromiseMultiple.toastPromiseMultiple}. */
+const TOAST_PROMISE_MULTIPLE_MESSAGES: toastPromiseMultiple.ToastPromiseMultipleMessages<backendModule.FileAsset> =
+    {
+        begin: expectedCount =>
+            `Deleting ${expectedCount} ${expectedCount === 1 ? 'file' : 'files'}...`,
+        inProgress: (successCount, expectedCount) =>
+            `Deleted ${successCount}/${expectedCount} ${expectedCount === 1 ? 'file' : 'files'}.`,
+        end: (successCount, expectedCount) =>
+            `Deleted ${successCount}/${expectedCount} ${expectedCount === 1 ? 'file' : 'files'}.`,
```

Comment:
done locally. it's worth noting that now the definition of every `TOAST_PROMISE_MULTIPLE_MESSAGES` is virtually identical, but i'd be cautious about refactoring that too, as it will likely result in a file being created for just the one function - unless you have any ideas?